io/subsystem/src/main/java/org/wildfly/extension/io/OutboundBindAddressResourceDefinition.java

```diff
@@ -49,8 +49,7 @@
         .setRestartAllServices()
         .build();
 
-    static final SimpleAttributeDefinition BIND_ADDRESS = new SimpleAttributeDefinitionBuilder("bind-address", ModelType.STRING)
-        .setRequired(true)
+    static final SimpleAttributeDefinition BIND_ADDRESS = new SimpleAttributeDefinitionBuilder("bind-address", ModelType.STRING, true)
```

Comment:
According to the XML schema this attribute is required: -

        <xs:attribute name="bind-address" type="xs:string" use="required">
            <annotation xmlns="http://www.w3.org/2001/XMLSchema">
                <documentation>
                    The bind address to use if the destination address matches.
                </documentation>
            </annotation>
        </xs:attribute>

By making this optional could mean the attribute is not written leaving the configuration with an error if validated against the schema.

Reply:
IMO that's up to the marsharaller and that is okay to make it optional in the model since there is a default value.

Reply:
@jamezp possibly but has anyone checked that the marshaller will always marshal the default here?

Reply:
I've not checked in this specific case no. IIRC the default is `true` though. I could be wrong and we'd definitely need to check that out first.